show validation loss during training

Feserich opened the pull request:
Hey, 
I enhanced the code, so it can show the validation loss additional to the training loss. 
This is optional and will be only displayed, if a test dataset is given to the fit() or fit_transform() method as argument (beside the training dataset). 
I think the validation loss is another important metric to evaluate the model. 
Please tell me if there are some enhancements to my code. 

Regards
Feserich

Feserich committed:
confusion of names

In stacked_autoencoder.py 'x_' is used for the decoded result, but the return from 'transform()' are the latent features not the decoded features.

Feserich committed:
show the global validation loss during training

Feserich committed:
remove folder

Feserich committed:
changed typo

rajarsheem commented on .idea/vcs.xml:
```diff
@@ -0,0 +1,6 @@
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+  </component>
+</project>
```
Can you please remove this file?

Feserich replied:
I removed that file in my following commit. The pull request contains 4 commits. 
In my latest commit I also changed the naming from eval_x to val_x.

rajarsheem commented on test.py:
```diff
@@ -12,5 +12,5 @@
 
 model = StackedAutoEncoder(dims=[200, 200], activations=['linear', 'linear'], epoch=[
                            3000, 3000], loss='rmse', lr=0.007, batch_size=100, print_step=200)
-model.fit(train_X)
```
Maybe rename test_X to valid_X ? Can you please keep both the usages (w/ and w/o valid_x) ?

Feserich replied:
Yes that's a good idea. 
Maybe also add an example for fit_transform()?

Feserich committed:
Upadated test.py with some examples

Feserich committed:
Bug: use clean input for the next layer

Feserich committed:
added Metric: R¹-Score

Feserich committed:
bugfix for R² Score

Feserich commented:
@rajarsheem I updated test.py with some examples. If you want a different naming or comment description feel free to edit it (you, as maintainer, can also edit my pull request).  
I also fixed a bug in my latest commit. 

Do you think it is better to only put noise on the training data and not on the validation data? Because after your model is trained and you want to use it,  you also don't put extra noise to your input data. So maybe the validation metrics without noise is more meaningful. What is your opinion?

Feserich committed:
bugfix for R² Score

Feserich commented:
@rajarsheem Do you know how to fix the Travis CI errors?

Feserich committed:
added a reconstruction function

Feserich committed:
added partial fit (save and reload train_op, weights and biases) untested!

Feserich committed:
create save path if not exist

Feserich committed:
cleanup

Feserich committed:
deleted R² Metric

Feserich committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	deepautoencoder/stacked_autoencoder.py
